app/store/credit.ts

```diff
@@ -0,0 +1,69 @@
+import { supabase } from "../components/hippo/supabase";
+import { StoreKey } from "../constant";
+import { createPersistStore } from "../utils/store";
+
+export const useCreditStore = createPersistStore(
+  { credit: 5 },
+  (set, get) => ({
+    async fetchCredit(userId: string, isAnonymous?: boolean) {
+      console.log("userId", userId);
+      let { data: creditQuery, error } = await supabase
+        .from("credits")
+        .select("*")
+        .eq("id", userId);
+      if (creditQuery?.length === 0) {
+        this.upsertCredit(userId, isAnonymous ? 5 : 200);
+      } else {
+        console.log("creditQuery", creditQuery);
+        set(() => ({
+          credit: creditQuery ? creditQuery[0]?.credit : 5,
+        }));
+      }
+    },
+
+    async upsertCredit(userId: string, creditNumber?: number) {
+      let { data: creditUpsert, error } = await supabase
+        .from("credits")
+        .upsert([
+          {
+            id: userId,
+            credit: creditNumber,
+          },
+        ])
+        .select();
+      console.log("creditUpsert", creditUpsert);
+      set(() => ({
+        credit: creditUpsert ? creditUpsert[0]?.credit : creditNumber,
+      }));
+    },
+
+    subcribe(userId: string) {
+      console.log("userId", userId);
+      return supabase
+        .channel("subcribe-credit")
```

Comment:
is there any limit on this?

Reply:
Em chưa đọc được thông tin nào về limit của nó realtime trên supabase ạ, nên e đoán là k có limit ạ